Modules/Core/ImageFunction/include/itkMeanImageFunction.hxx

```diff
@@ -47,27 +51,28 @@ MeanImageFunction<TInputImage, TCoordRep>::EvaluateAtIndex(const IndexType & ind
     return (NumericTraits<RealType>::max());
   }
 
-  // Create an N-d neighborhood kernel, using a zeroflux boundary condition
-  typename InputImageType::SizeType kernelSize;
-  kernelSize.Fill(m_NeighborhoodRadius);
-
-  ConstNeighborhoodIterator<InputImageType> it(
-    kernelSize, this->GetInputImage(), this->GetInputImage()->GetBufferedRegion());
-
-  // Set the iterator at the desired location
-  it.SetLocation(index);
+  const Experimental::ShapedImageNeighborhoodRange<const InputImageType> neighborhoodRange(
+    *image, index, m_NeighborhoodOffsets);
 
   // Walk the neighborhood
-  const unsigned int size = it.Size();
-  for (unsigned int i = 0; i < size; ++i)
+  for (const InputPixelType pixelValue : neighborhoodRange)
   {
-    sum += static_cast<RealType>(it.GetPixel(i));
+    sum += static_cast<RealType>(pixelValue);
   }
-  sum /= double(it.Size());
 
-  return sum;
+  return sum / static_cast<double>(neighborhoodRange.size());
 }
 
+
+template <typename TInputImage, typename TCoordRep>
+void
+MeanImageFunction<TInputImage, TCoordRep>::SetNeighborhoodRadius(const unsigned int radius)
+{
+  m_NeighborhoodOffsets = Experimental::GenerateRectangularImageNeighborhoodOffsets(ImageSizeType::Filled(radius));
```

Comment:
Here we want to compare to the previous value of `radius` and call `this->Modified()` if son, per:

https://github.com/InsightSoftwareConsortium/ITK/blob/8e4ab68c240f439a0b188796794f2c7066ea8a46/Modules/Core/Common/include/itkMacro.h#L875-L879

Reply:
@thewtex Thanks, good catch, Matt! I just did a force-push , having added a compare to the previous radius value and `this->Modified()` call. Now I see, `itkSetMacro` declares the function `virtual`: https://github.com/InsightSoftwareConsortium/ITK/blob/8e4ab68c240f439a0b188796794f2c7066ea8a46/Modules/Core/Common/include/itkMacro.h#L871 

Do you still want `SetNeighborhoodRadius` to be `virtual`? Personally I don't think it's necessary. You know, we have put a lot of effort into removing `virtual` keywords from the past!